src/test/util/setup_common.cpp

```diff
@@ -98,9 +98,20 @@ std::ostream& operator<<(std::ostream& os, const uint256& num)
 }
 
 BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::vector<const char*>& extra_args)
-    : m_path_root{fs::temp_directory_path() / "test_common_" PACKAGE_NAME / g_insecure_rand_ctx_temp_path.rand256().ToString()},
-      m_args{}
+    : m_args{}
 {
+    char* test_path_env{std::getenv("BITCOIN_TEST_PATH")};
+    std::string test_path;
+    if (test_path_env == nullptr) {
+        test_path = g_insecure_rand_ctx_temp_path.rand256().ToString();
+    } else {
+        test_path = test_path_env;
+        if (fs::PathFromString(test_path).is_absolute()) {
```

Comment:
Maybe just use a hardcoded path item like `"fixed"` and turn the env into a flag? This should be fine, because anyone can still modify the root temp dir.

Reply:
I know this sounds a little far-fetched, but I have actually done the following: I have two test binaries, one from a PR branch and the other from master, and the test fails on the PR branch but passes on master. I've single-stepped through the tests simultaneously (in sync) comparing them to see what happens leading up to the failure. In this case, it would be convenient to have two different fixed data directories (if I needed to restart the test many times) because I also want to watch the two `debug.log` files.

Also, as long as we're going to the trouble of implementing and documenting an environment variable, it's no extra work to make it a value instead of a flag.

Reply:
Yes, that is still possible. You can set `TMPDIR=/tmp/1`, see https://en.cppreference.com/w/cpp/filesystem/temp_directory_path

Not sure if relevant, but `BITCOIN_TEST_PATH=../../home/user` will still delete your home dir.

Reply:
Good catch! I just force-pushed an update to require that `BITCOIN_TEST_PATH` doesn't contain the separator character. That should take care of both problems. I would say that if the user has set `TMPDIR`, it should be safe to delete the given directory, even if it's within the user's home directory, since paths containing `../` are no longer allowed.
```
$ BITCOIN_TEST_PATH=../mydatadir src/test/test_bitcoin --run_test=getarg_tests/doubledash
Running 1 test case...
unknown location(0): fatal error: in "getarg_tests/doubledash": std::runtime_error: BITCOIN_TEST_PATH cannot contain path separators
test/getarg_tests.cpp(380): last checkpoint: "doubledash" fixture ctor

*** 1 failure is detected in the test module "Bitcoin Core Test Suite"
$
```